slick/src/main/scala/slick/util/AsyncExecutor.scala

```diff
@@ -70,15 +104,21 @@ object AsyncExecutor extends Logging {
         if(!executor.awaitTermination(30, TimeUnit.SECONDS))
           logger.warn("Abandoning ThreadPoolExecutor (not yet destroyed after 30 seconds)")
       }
+
     }
   }
 
   def default(name: String = "AsyncExecutor.default"): AsyncExecutor =
     apply(name, 20, 1000)
 
+  sealed trait Priority
+  case object LowPriority extends Priority
+  case object MediumPriority extends Priority
+  case object HighPriority extends Priority
 
   trait PrioritizedRunnable extends Runnable {
-    def highPriority: Boolean
+    def priority: Priority
+    @volatile var connectionReleased = false
```

Comment:
Does this have to be `@volatile`? It's only modified from code running on the database thread pool and read in `afterExecute` which runs on the same thread.